diff --git a/core/pathmap.mk b/core/pathmap.mk
index 3f38a6b..f4f9671 100644
--- a/core/pathmap.mk
+++ b/core/pathmap.mk
@@ -81,6 +81,7 @@ JNI_H_INCLUDE := $(call include-path-for,libnativehelper)/nativehelper
 # is so common and the other framework libraries can have mutual
 # interdependencies.
 #
+# BEGIN privacy-added: privacy
 FRAMEWORKS_BASE_SUBDIRS := \
 	$(addsuffix /java, \
 	    core \
@@ -92,6 +93,7 @@ FRAMEWORKS_BASE_SUBDIRS := \
 	    media/mca/filterpacks \
 	    drm \
 	    opengl \
+	    privacy \
 	    sax \
 	    telephony \
 	    wifi \
@@ -100,6 +102,7 @@ FRAMEWORKS_BASE_SUBDIRS := \
 	    voip \
 	    fmradio \
 	 )
+# END privacy-added: privacy
 
 #
 # A version of FRAMEWORKS_BASE_SUBDIRS that is expanded to full paths from
diff --git a/target/product/core.mk b/target/product/core.mk
index da1abae..8d3e087 100644
--- a/target/product/core.mk
+++ b/target/product/core.mk
@@ -18,6 +18,7 @@ PRODUCT_BRAND := generic
 PRODUCT_DEVICE := generic
 PRODUCT_NAME := core
 
+# BEGIN privacy-added: PDroid
 PRODUCT_PACKAGES += \
     ApplicationsProvider \
     BackupRestoreConfirmation \
@@ -32,6 +33,7 @@ PRODUCT_PACKAGES += \
     Home \
     KeyChain \
     MediaProvider \
+    PDroid \
     PackageInstaller \
     PhotoTable \
     PicoTts \
@@ -130,6 +132,7 @@ PRODUCT_PACKAGES += \
     zoneinfo.dat \
     zoneinfo.idx \
     zoneinfo.version
+# END privacy-added: PDroid
 
 PRODUCT_COPY_FILES += \
     system/core/rootdir/init.usb.rc:root/init.usb.rc \
